GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1283,17 +1286,16 @@ private void UnstageAllToolStripMenuItemClick(object sender, EventArgs e)
                 lastSelection = _currentSelection;
             }
 
-            Action stageAreaLoaded = null;
-            stageAreaLoaded = () =>
+            void StageAreaLoaded()
             {
                 _currentFilesList = Unstaged;
                 RestoreSelectedFiles(Unstaged.GitItemStatuses, Staged.GitItemStatuses, lastSelection);
                 Unstaged.Focus();
 
-                OnStageAreaLoaded -= stageAreaLoaded;
-            };
+                OnStageAreaLoaded -= StageAreaLoaded;
```

Comment:
❗️ This won't work. The delegate used for registering and unregistering the event is no longer the same. All event registrations that switched to local functions should be reverted.

Reply:
Are you sure about this? I've seen the bug you're talking about but I cannot trigger it in a scenario like this in a unit test. Here is a simple test that shows the same scenario passing:

```c#
private event Action ActionEvent;

[Test]
public void LocalFunctionLambdaTest()
{
    Assert.Null(ActionEvent);

    ActionEvent += Handler;

    Assert.NotNull(ActionEvent);

    ActionEvent();

    Assert.Null(ActionEvent);

    void Handler()
    {
        ActionEvent -= Handler;
    }
}
```

I've tried introducing closures and other delegate signatures but without success. My guess is the compiler closes over the reference to the instance of the local function. Will play with this some more, but so far I can't find a need to revert anything here.

Reply:
👍 I believe you are correct. The big problem occurs when the code relies on two delegates having the same form:

```csharp
Event += (sender, e) => { };
Event -= (sender, e) => { }; // May or may not unregister
```

If this code, you only have one definition of the functions.